android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -1013,8 +1013,13 @@ private void peerConnectionGetStatsAsync(String trackId,
     }
 
     @ReactMethod
-    public void peerConnectionClose(int id) {
-        ThreadUtils.runOnExecutor(() -> peerConnectionCloseAsync(id));
+    public void peerConnectionClose(int id, Callback cb) {
+        ThreadUtils.runOnExecutor(() -> {
+            peerConnectionCloseAsync(id);
```

Comment:
Is this call actually async? If so, do we need to pass the callback further down or is it safe to be called at this level?

Reply:
@streamer45 I don't think this call is actually Async. At least, it's not async in the android code. I looked at the c++ code, and if I found the right code (the Android code calls `nativeClose()` which I can't find, but I'm guessing that's being translated to `close()` in the peerconnection c++ code) then it looks like it's all blocking calls to other threads (according to the comments). Either way, the android code is written as if it were synchronous, so I'm not sure why they call that method  `peerConnectionCloseAsync`.
So that's my reasoning for stopping the callback here -- if the native code really is async, then there's nothing we can do about it in the android code, and there shouldn't be any expectation that we should wait (since, if it were async, there's no way for us to know when the native code is finished).

Hope that clears it up?

Reply:
Yep, that's good enough for me. Thanks :+1: